usr/lib/icsf_stdll/icsf.c

```diff
@@ -876,14 +876,40 @@ static int is_numeric_attr(CK_ULONG type)
     case CKA_CLASS:
     case CKA_KEY_TYPE:
     case CKA_CERTIFICATE_TYPE:
+    case CKA_HW_FEATURE_TYPE:
     case CKA_KEY_GEN_MECHANISM:
     case CKA_VALUE_LEN:
+    case CKA_VALUE_BITS:
     case CKA_MODULUS_BITS:
+    case CKA_PRIME_BITS:
+    case CKA_SUBPRIME_BITS:
+    case CKA_CERTIFICATE_CATEGORY:
+    case CKA_JAVA_MIDP_SECURITY_DOMAIN:
         return 1;
     }
     return 0;
 }
 
+static CK_RV find_ulong_attribute(CK_ATTRIBUTE *attrs, CK_ULONG attrs_len,
+                                  CK_ATTRIBUTE_TYPE type, CK_ULONG *value)
+{
+    CK_ULONG i;
```

Comment:
size_t instead of CK_ULONG?

Reply:
Why ? `i` is used to iterate over the attributes up to `attrs_len`. `attrs_len` is also a CK_ULONG, so `i` should be the same type.

Reply:
ok good